File: questapp-backend/src/main/java/com/example/questappbackend/services/abs/LikeService.java
package com.example.questappbackend.services.abs;


import com.example.questappbackend.dto.LikeDtoForCreate;
import com.example.questappbackend.dto.responses.LikeDtoResponse;
import com.example.questappbackend.entities.Like;

import java.util.List;
import java.util.Optional;

public interface LikeService  {

    List<LikeDtoResponse> getAllLikesWithParam(Optional<Long> userId, Optional<Long> postId);

    Like getOneLike (Long likeId);

    Like createOneLike (LikeDtoForCreate request);

    void deleteOneLike (Long likeId);
}

File: questapp-frontend/src/service/LikeService.js
import axios from "axios";

const base_url = "http://localhost:8080/likes";

class LikeService {
  async createOneLike(like) {
    return await axios.post(base_url, like);
  }

  async deleteOneLike(likeId) {
    const url = `${base_url}/${likeId}`;
    axios.delete(url);
  }
}

export default new LikeService();
